integrationtests/helmops/controller/controller_test.go

```diff
@@ -0,0 +1,972 @@
+	labels := make(map[string]string)
+	for k, v := range m {
+		s, ok := v.(string)
+		if ok {
+			labels[k] = s
+		}
+	}
+	return labels
+}
+
+func randHelmOptions() *fleet.HelmOptions {
+	// we always have helm options in HelmApp resources
+	h := &fleet.HelmOptions{
+		Chart:                   randString(),
+		Repo:                    randString(),
+		ReleaseName:             randString(),
+		Version:                 randString(), // return also semver version?
+		TimeoutSeconds:          rand.Intn(3),
+		Values:                  &fleet.GenericMap{Data: randInterfaceMap()},
+		Force:                   randBool(),
+		TakeOwnership:           randBool(),
+		MaxHistory:              rand.Intn(4),
+		ValuesFiles:             randStringSlice(),
+		WaitForJobs:             randBool(),
+		Atomic:                  randBool(),
+		DisablePreProcess:       randBool(),
+		DisableDNS:              randBool(),
+		SkipSchemaValidation:    randBool(),
+		DisableDependencyUpdate: randBool(),
+	}
+
+	return h
+}
+
+func randKustomizeOptions() *fleet.KustomizeOptions {
+	if randBool() {
+		return nil
+	}
+	o := &fleet.KustomizeOptions{}
+	o.Dir = randString()
+	return o
+}
+
+func randBundleDeploymentOptions() fleet.BundleDeploymentOptions {
+	o := fleet.BundleDeploymentOptions{
+		DefaultNamespace: randString(),
+		TargetNamespace:  randString(),
+		Kustomize:        randKustomizeOptions(),
+		Helm:             randHelmOptions(),
+		CorrectDrift:     randCorrectDrift(),
+		ServiceAccount:   randString(),
+	}
+	return o
+}
+
+func randCorrectDrift() *fleet.CorrectDrift {
+	if randBool() {
+		return nil
+	}
+	r := &fleet.CorrectDrift{
+		Enabled:         randBool(),
+		Force:           randBool(),
+		KeepFailHistory: randBool(),
+	}
+
+	return r
+}
+
+func getRandomHelmAppWithTargets(name string, t []fleet.BundleTarget) fleet.HelmApp {
+	namespace = testenv.NewNamespaceName(
+		name,
+		rand.New(rand.NewSource(time.Now().UnixNano())),
+	)
+	h := fleet.HelmApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		// add a few random values
+		Spec: fleet.HelmAppSpec{
+			Labels: randStringMap(),
+			BundleSpec: fleet.BundleSpec{
+				BundleDeploymentOptions: randBundleDeploymentOptions(),
+			},
+			HelmSecretName:        randString(),
+			InsecureSkipTLSverify: randBool(),
+		},
+	}
+
+	h.Spec.Targets = t
+
+	return h
+}
+
+// compareBundleAndHelmAppSpecs compares the part that it is expected to be equal
+// between a Bundle's spec and a HelmApp's spec.
+func compareBundleAndHelmAppSpecs(g Gomega, bundle fleet.BundleSpec, helmapp fleet.BundleSpec) {
+	g.Expect(bundle.BundleDeploymentOptions).To(Equal(helmapp.BundleDeploymentOptions))
+	g.Expect(bundle.Paused).To(Equal(helmapp.Paused))
+	g.Expect(bundle.RolloutStrategy).To(Equal(helmapp.RolloutStrategy))
+	g.Expect(bundle.Resources).To(Equal(helmapp.Resources))
+	g.Expect(bundle.Targets).To(Equal(helmapp.Targets))
+	g.Expect(bundle.Targets).To(Equal(helmapp.Targets))
```

Comment:
This line is duplicated. Was another field intended to be compared here, eg. `ContentsID`?

Reply:
No, it was just duplicated. Deleted